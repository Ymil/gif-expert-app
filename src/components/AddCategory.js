import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState("");

    const handlerInputChange = (e) => {    
        setInputValue(e.target.value);
    }

    const handlerSumit = (e) => {
        if(inputValue.trim().length > 2){
            setCategories(categories => [inputValue, ...categories]);
            setInputValue('')
        }
        
        
        e.preventDefault()
    }

    return (
        <form onSubmit={ handlerSumit }>         
            <input 
            type='text'
            value={inputValue}
            onChange={ handlerInputChange }
            placeholder='Ingrese la categoria a buscar'
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
