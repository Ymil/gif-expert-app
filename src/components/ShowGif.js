import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const ShowGif = ({title, url}) => {
    
    const [state, setstate] = useState(false);

    return (
        <div className='card animate__animated animate__pulse'>
            { !state && <p className="animate__animated animate__flash animate__infinite">Cargando...</p>}
            <img src={url} onLoad={() => setstate(true)}/>
            <p className='title'>{title.trim().length == 0 ? "Sin titulo" : title}</p>
        </div>
    )
}

ShowGif.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}