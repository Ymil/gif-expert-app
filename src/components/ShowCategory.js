import PropTypes from 'prop-types'
import { ShowGif } from './ShowGif'
import { useFetchGifs } from './hooks/useFetchGifs'
import "animate.css"
export const ShowCategory = ({category}) => {
    const {gifs, loading} = useFetchGifs(category);
    
    return (
        <>
            <h2>{category}</h2>

            {
                loading && <p className="animate__animated animate__flash">Cargando...</p>
            }
            <div className="card-grid">
                { gifs.map(
                    (gif) => <ShowGif {...gif} key={gif.id} />
                )}
            </div>
        </>
    )
}

ShowCategory.propTypes = {
    category: PropTypes.string.isRequired
}


