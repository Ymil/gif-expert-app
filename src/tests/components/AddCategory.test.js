import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe('Probando AddCategory', () => {
    
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach( () => {
        jest.clearAllMocks() 
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });

    test('should ', () => {
        expect(wrapper).toMatchSnapshot()
        
    });

    test('debe cambiar caja de texto', () => {
        let input = wrapper.find("input");
        const value = "hola mundo";
        input.simulate("change", {
            target:{
                value: value
            }
        });
        input = wrapper.find("input");
        expect(input.prop('value')).toBe(value);
    })
    
    test('disparando submit sin postear informacion', () => {        
        const form = wrapper.find("form");

        form.simulate("submit", {
            preventDefault: () => {}
        })

        expect(setCategories).not.toHaveBeenCalled()
    })


    test('verificando reset de formulario despues de submit', () => {
        let input = wrapper.find("input");
        const value = "hola mundo";
        input.simulate("change", {
            target:{
                value: value
            }
        });
        
        const form = wrapper.find("form");

        form.simulate("submit", {
            preventDefault: () => {}
        })

        expect(setCategories).toHaveBeenCalled()
        expect(setCategories).toHaveBeenCalledWith(
            expect.any(Function)
        );

        input = wrapper.find("input");
        expect(input.prop('value')).toBe("");
    })
    
    
})
