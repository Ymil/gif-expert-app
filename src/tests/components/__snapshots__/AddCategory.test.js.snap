// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando AddCategory should  1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Ingrese la categoria a buscar"
    type="text"
    value=""
  />
</form>
`;
